dokka-subprojects/plugin-base/src/main/resources/dokka/scripts/navigation-loader.js

```diff
@@ -1,95 +1,140 @@
 /*
- * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ * Copyright 2014-2025 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
+const TOC_STATE_KEY_PREFIX = 'TOC_STATE::';
+const TOC_CONTAINER_ID = 'sideMenu';
+const TOC_SCROLL_CONTAINER_ID = 'leftColumn';
+const TOC_PART_CLASS = 'toc--part';
+const TOC_PART_HIDDEN_CLASS = 'toc--part_hidden';
+const TOC_LINK_CLASS = 'toc--link';
 
-navigationPageText = fetch(pathToRoot + "navigation.html").then(response => response.text())
+(function () {
+  function displayToc() {
```

Comment:
this is the part with the core ToC logics. It's loaded in a separate script to make sure that we load navigation as soon as possible